From a5a7850119075da8224a24d2d1f1ab14a667b996 Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@gmail.com>
Date: Mon, 29 Jul 2024 12:48:47 -0700
Subject: [PATCH 2/9] add gpus with gfx-name to build list

add all rocm sdk gpu's to build list
and use the gfx* name for them instead
of marketing name. This allows using the
same logic for passing the gpulist to
aotrition that what is used for building all
other rocm libraries and apps.

Signed-off-by: Mika Laitio <lamikr@gmail.com>
---
 include/aotriton/util.h | 10 ++++++++++
 v2python/compile.py     | 14 ++++++++++++++
 v2python/gpu_targets.py | 42 +++++++++++++++++++++++++++++++++++++++++
 v2src/util.cc           | 10 ++++++++++
 4 files changed, 76 insertions(+)

diff --git a/include/aotriton/util.h b/include/aotriton/util.h
index efb9395..74033e3 100644
--- a/include/aotriton/util.h
+++ b/include/aotriton/util.h
@@ -37,8 +37,18 @@ enum GpuArch : uint64_t {
   GPU_ARCH_UNKNOWN = 0,
   GPU_ARCH_AMD_GFX90A = CAT(GpuVendor::kAMD, 0x90a),
   GPU_ARCH_AMD_GFX942 = CAT(GpuVendor::kAMD, 0x942),
+  GPU_ARCH_AMD_GFX1010 = CAT(GpuVendor::kAMD, 0x1010),
+  GPU_ARCH_AMD_GFX1011 = CAT(GpuVendor::kAMD, 0x1011),
+  GPU_ARCH_AMD_GFX1012 = CAT(GpuVendor::kAMD, 0x1012),
+  GPU_ARCH_AMD_GFX1030 = CAT(GpuVendor::kAMD, 0x1030),
+  GPU_ARCH_AMD_GFX1031 = CAT(GpuVendor::kAMD, 0x1031),
+  GPU_ARCH_AMD_GFX1032 = CAT(GpuVendor::kAMD, 0x1032),
+  GPU_ARCH_AMD_GFX1035 = CAT(GpuVendor::kAMD, 0x1035),
+  GPU_ARCH_AMD_GFX1036 = CAT(GpuVendor::kAMD, 0x1036),
   GPU_ARCH_AMD_GFX1100 = CAT(GpuVendor::kAMD, 0x1100),
   GPU_ARCH_AMD_GFX1101 = CAT(GpuVendor::kAMD, 0x1101),
+  GPU_ARCH_AMD_GFX1102 = CAT(GpuVendor::kAMD, 0x1102),
+  GPU_ARCH_AMD_GFX1103 = CAT(GpuVendor::kAMD, 0x1103),
 };
 
 template<int Rank>
diff --git a/v2python/compile.py b/v2python/compile.py
index 78dd4de..81d928b 100644
--- a/v2python/compile.py
+++ b/v2python/compile.py
@@ -20,6 +20,20 @@ KNOWN_TARGETS = {
         'MI300X' : GPUTarget('hip', 'gfx942', 64),
         'Navi31' : GPUTarget('hip', 'gfx1100', 32),
         'Navi32' : GPUTarget('hip', 'gfx1101', 32),
+        'gfx90a' : GPUTarget('hip', 'gfx90a', 64),
+        'gfx942' : GPUTarget('hip', 'gfx942', 64),
+        'gfx1010' : GPUTarget('hip', 'gfx1010', 32),
+        'gfx1011' : GPUTarget('hip', 'gfx1011', 32),
+        'gfx1012' : GPUTarget('hip', 'gfx1012', 32),
+        'gfx1030' : GPUTarget('hip', 'gfx1030', 32),
+        'gfx1031' : GPUTarget('hip', 'gfx1031', 32),
+        'gfx1032' : GPUTarget('hip', 'gfx1032', 32),
+        'gfx1035' : GPUTarget('hip', 'gfx1035', 32),
+        'gfx1036' : GPUTarget('hip', 'gfx1036', 32),
+        'gfx1100' : GPUTarget('hip', 'gfx1100', 32),
+        'gfx1101' : GPUTarget('hip', 'gfx1101', 32),
+        'gfx1102' : GPUTarget('hip', 'gfx1102', 32),
+        'gfx1103' : GPUTarget('hip', 'gfx1103', 32),
 }
 
 desc = """
diff --git a/v2python/gpu_targets.py b/v2python/gpu_targets.py
index fb51fef..32ccfff 100644
--- a/v2python/gpu_targets.py
+++ b/v2python/gpu_targets.py
@@ -6,6 +6,20 @@ AOTRITON_SUPPORTED_GPUS = {
     'MI300X'    : 'GPU_ARCH_AMD_GFX942',
     'Navi31'    : 'GPU_ARCH_AMD_GFX1100',
     'Navi32'    : 'GPU_ARCH_AMD_GFX1101',
+    'gfx90a'    : 'GPU_ARCH_AMD_GFX90A',
+    'gfx942'    : 'GPU_ARCH_AMD_GFX942',
+    'gfx1010'   : 'GPU_ARCH_AMD_GFX1010',
+    'gfx1011'   : 'GPU_ARCH_AMD_GFX1011',
+    'gfx1012'   : 'GPU_ARCH_AMD_GFX1012',
+    'gfx1030'   : 'GPU_ARCH_AMD_GFX1030',
+    'gfx1031'   : 'GPU_ARCH_AMD_GFX1031',
+    'gfx1032'   : 'GPU_ARCH_AMD_GFX1032',
+    'gfx1035'   : 'GPU_ARCH_AMD_GFX1035',
+    'gfx1036'   : 'GPU_ARCH_AMD_GFX1036',
+    'gfx1100'   : 'GPU_ARCH_AMD_GFX1100',
+    'gfx1101'   : 'GPU_ARCH_AMD_GFX1101',
+    'gfx1102'   : 'GPU_ARCH_AMD_GFX1102',
+    'gfx1103'   : 'GPU_ARCH_AMD_GFX1103',
 }
 
 AOTRITON_GPU_ARCH_TUNING_STRING = {
@@ -13,6 +27,20 @@ AOTRITON_GPU_ARCH_TUNING_STRING = {
     'MI300X'    : 'gfx942',
     'Navi31'    : 'gfx1100',
     'Navi32'    : 'gfx1101',
+    'gfx90a'    : 'gfx90a',
+    'gfx942'    : 'gfx942',
+    'gfx1010'   : 'gfx1010',
+    'gfx1011'   : 'gfx1011',
+    'gfx1012'   : 'gfx1012',
+    'gfx1030'   : 'gfx1030',
+    'gfx1031'   : 'gfx1031',
+    'gfx1032'   : 'gfx1032',
+    'gfx1035'   : 'gfx1035',
+    'gfx1036'   : 'gfx1036',
+    'gfx1100'   : 'gfx1100',
+    'gfx1101'   : 'gfx1101',
+    'gfx1102'   : 'gfx1102',
+    'gfx1103'   : 'gfx1103',
 }
 
 AOTRITON_GPU_WARPSIZE = {
@@ -20,4 +48,18 @@ AOTRITON_GPU_WARPSIZE = {
     'MI300X'    : 64,
     'Navi31'    : 32,
     'Navi32'    : 32,
+    'gfx90a'    : 64,
+    'gfx942'    : 64,
+    'gfx1010'   : 32,
+    'gfx1011'   : 32,
+    'gfx1012'   : 32,
+    'gfx1030'   : 32,
+    'gfx1031'   : 32,
+    'gfx1032'   : 32,
+    'gfx1035'   : 32,
+    'gfx1036'   : 32,
+    'gfx1100'   : 32,
+    'gfx1101'   : 32,
+    'gfx1102'   : 32,
+    'gfx1103'   : 32,
 }
diff --git a/v2src/util.cc b/v2src/util.cc
index 2c47e1d..2a6128b 100644
--- a/v2src/util.cc
+++ b/v2src/util.cc
@@ -30,8 +30,18 @@ private:
 std::unordered_map<std::string, GpuArch> LazyArch::string_to_arch = {
   {"gfx90a:sramecc+:xnack-", GPU_ARCH_AMD_GFX90A},
   {"gfx942:sramecc+:xnack-", GPU_ARCH_AMD_GFX942},
+  {"gfx1010", GPU_ARCH_AMD_GFX1010},
+  {"gfx1011", GPU_ARCH_AMD_GFX1011},
+  {"gfx1012", GPU_ARCH_AMD_GFX1012},
+  {"gfx1030", GPU_ARCH_AMD_GFX1030},
+  {"gfx1031", GPU_ARCH_AMD_GFX1031},
+  {"gfx1032", GPU_ARCH_AMD_GFX1032},
+  {"gfx1035", GPU_ARCH_AMD_GFX1035},
+  {"gfx1036", GPU_ARCH_AMD_GFX1036},
   {"gfx1100", GPU_ARCH_AMD_GFX1100},
   {"gfx1101", GPU_ARCH_AMD_GFX1101},
+  {"gfx1102", GPU_ARCH_AMD_GFX1102},
+  {"gfx1103", GPU_ARCH_AMD_GFX1103},
 };
 
 GpuArch
-- 
2.48.1

